app/src/main/java/tttTDD/Java/ConsoleClient.java

```diff
@@ -105,13 +106,52 @@ private int convertLocationToInput(int value) {
     private void renderBoard() {
         String[] board = game.getBoard();
         int boardSize = gameConfig.getBoardSizeConfiguration();
+        String maxInput = String.valueOf(boardSize * boardSize);
+
+        String rowWithLines;
+        ArrayList<String> fullBoard = new ArrayList<>();
```

Comment:
I think this code is a bit hard to follow and could be cleaner. But I'm not sure how it actually looks.

An ArrayList feels like an inefficient data structure here. When building strings, it's recommended to use the StringBuilder class, which might help keep things a bit cleaner since it provides a nice interface for some of the things you're doing here (instead of calling `join` on the string array)

https://docs.oracle.com/javase/8/docs/api/java/lang/StringBuilder.html

Reply:
ooh, nice